services/src/main/java/io/druid/cli/PullDependencies.java

```diff
@@ -18,34 +18,103 @@
 package io.druid.cli;
 
 import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableList;
 import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
 import com.google.inject.Inject;
+import com.metamx.common.ISE;
+import com.metamx.common.StringUtils;
+import com.metamx.common.logger.Logger;
 import io.airlift.airline.Command;
 import io.airlift.airline.Option;
 import io.druid.guice.ExtensionsConfig;
 import io.druid.indexing.common.config.TaskConfig;
-import io.druid.initialization.Initialization;
+import io.tesla.aether.Repository;
 import io.tesla.aether.internal.DefaultTeslaAether;
+import org.apache.commons.io.FileUtils;
+import org.eclipse.aether.artifact.Artifact;
+import org.eclipse.aether.artifact.DefaultArtifact;
+import org.eclipse.aether.collection.CollectRequest;
+import org.eclipse.aether.graph.Dependency;
+import org.eclipse.aether.graph.DependencyFilter;
+import org.eclipse.aether.graph.DependencyNode;
+import org.eclipse.aether.resolution.DependencyRequest;
+import org.eclipse.aether.util.artifact.JavaScopes;
+import org.eclipse.aether.util.filter.DependencyFilterUtils;
 
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URISyntaxException;
 import java.util.List;
+import java.util.Set;
 
 
 @Command(
     name = "pull-deps",
-    description = "Pull down dependencies to the local repository specified by druid.extensions.localRepository"
+    description = "Pull down dependencies to the local repository specified by druid.extensions.localRepository, extensions directory specified by druid.extensions.extensionsDir and hadoop depenencies directory specified by druid.extensions.hadoopDependenciesDir"
 )
 public class PullDependencies implements Runnable
 {
-  @Option(name = {"-c", "--coordinate"},
-          title = "coordinate",
-          description = "extra dependencies to pull down (e.g. hadoop coordinates)",
-          required = false)
-  public List<String> coordinates;
+  private static final Logger log = new Logger(PullDependencies.class);
 
-  @Option(name = "--no-default-hadoop",
-          description = "don't pull down the default HadoopIndexTask dependencies",
-          required = false)
-  public boolean noDefaultHadoop;
+  private static final Set<String> exclusions = Sets.newHashSet(
+      "io.druid",
+      "com.metamx.druid"
+  );
+
+  @Option(
+      name = {"-c", "--coordinate"},
+      title = "coordinate",
+      description = "Extension coordinate to pull down, followed by a maven coordinate, e.g. io.druid.extensions:mysql-metadata-storage:0.8.0-rc1",
```

Comment:
err, can we only do stables or omit it altogether i.e. just write <version> instead of an actual version